docker/Dockerfile

```
@@ -0,0 +1,50 @@
+# ARGs obtained from the compose.yml file
+
+ARG BASE_OS
+ARG RUST_VERSION
+
+FROM rust:${RUST_VERSION}-${BASE_OS}
+
+# Nodejs installation
+
+RUN ["mkdir", "/usr/local/node"]
+WORKDIR /usr/local/node
+
+## Retrieves the targeted version of Node in the container
+
+COPY .nvmrc /tmp/.nvmrc
+
+## Download the Node.js standelone binary archive && extract it && clean files
+
+RUN ["/bin/sh", "-c", "NODE_VERSION=$(cat /tmp/.nvmrc) && \
+    wget https://nodejs.org/dist/v${NODE_VERSION}/node-v${NODE_VERSION}-linux-x64.tar.xz && \
+    tar xf node-v${NODE_VERSION}-linux-x64.tar.xz --strip-components 1 && \
+    rm node-v${NODE_VERSION}-linux-x64.tar.xz /tmp/.nvmrc"]
+
+## Ensure Node.js binaries are available in the PATH
+
+ENV PATH /usr/local/node/bin:$PATH
+
+# pnpm installation
+
+RUN ["npm", "install", "-g", "pnpm"]
```

I just you install pnpm via `npm` this is actually installing the last version of `pnpm`  which is 9.15.2. On the project we are on 9.14.2.

This might be an issue when pnpm releases a new major version.

I think that node [corepack](https://nodejs.org/docs/latest-v22.x/api/corepack.html) could help us here:

```suggestion
RUN ["corepack ", "install"]
```

This command should install the package manager configured in the local project (`package.json#packageManager` entry)

---

Ill try this tomorrow !

---

Features I didn't know. thank for share it :D